pages/admin/invite/userAccountPage/[id].tsx

```diff
@@ -6,13 +6,256 @@ import Combobox from "components/Combobox";
 import { User } from "@prisma/client";
 import Button from "components/Button";
 import { UpdateUserDTO } from "pages/api/admin/users/update";
-import { UserRoleType, IUser, UserStatus } from "interfaces";
+import { UserRoleType, IUser, UserRoleLabel, UserStatus } from "interfaces";
+import FormField from "components/FormField";
+import Joi from "joi";
+import { joiResolver } from "@hookform/resolvers/joi";
+import { useForm } from "react-hook-form";
+import Icon from "components/Icon";
 
-const UserAccountPage: React.FunctionComponent<UserRequest> = ({
-  onAccept,
+interface AdminEditUserFormValues {
+  firstName: string;
+  lastName: string;
+  email: string;
+  phoneNumber?: string;
+  role: UserRoleType;
+}
+
+interface EditUserProps {
+  user?: IUser;
+  isEditing: boolean;
+  setIsEditing: React.Dispatch<React.SetStateAction<boolean>>;
+  setUser: (user: IUser) => void;
+}
+
+type ModalProps = React.PropsWithChildren<{
+  open?: boolean;
+}>;
+
+const Modal: React.FC<ModalProps> = ({ children, open }: ModalProps) => {
+  return open ? (
+    <div className="absolute top-0 left-0 w-screen h-screen bg-dark bg-opacity-50 flex justify-center items-center">
+      <div className="bg-white rounded w-3/4 px-10 pt-12 pb-8">{children}</div>
+    </div>
+  ) : null;
+};
+
+const AdminEditUserFormSchema = Joi.object<AdminEditUserFormValues>({
+  firstName: Joi.string().trim().required(),
+  lastName: Joi.string().trim().required(),
+  email: Joi.string()
+    .trim()
+    .email({ tlds: { allow: false } })
+    .required(),
+  phoneNumber: Joi.string().optional(),
+  role: Joi.string()
+    .valid(...Object.values(UserRoleType))
+    .required(),
+});
+
+const EditUser: React.FunctionComponent<EditUserProps> = ({
+  user,
+  isEditing,
+  setIsEditing,
+  setUser,
 }) => {
+  const [submitting, setSubmitting] = useState(false);
+  const [error, setError] = useState("");
+  const [currRole, setCurrRole] = useState<UserRoleType>();
+  const { errors, register, handleSubmit } = useForm<AdminEditUserFormValues>({
+    resolver: joiResolver(AdminEditUserFormSchema),
+  });
+
+  useEffect(() => {
+    setCurrRole(user?.defaultRole.type);
+  }, [user]);
+
+  async function onSubmit(
+    values: AdminEditUserFormValues,
+    event?: React.BaseSyntheticEvent
+  ): Promise<void> {
+    event?.preventDefault();
+    if (submitting) {
+      return;
+    }
+    try {
+      const response = await fetch(`/api/admin/users/${user?.id}`, {
+        method: "PATCH",
+        headers: { "Content-Type": "application/json" },
+        credentials: "include",
+        body: JSON.stringify({
+          email: values.email,
+          name: `${values.firstName} ${values.lastName}`,
+          phoneNumber: values.phoneNumber,
+          roles: [values.role as UserRoleType],
+        } as UpdateUserDTO),
+      });
```

Comment:
There is a type error that is thrown here, which is probably why the update for editing is not functioning properly.
Error:
`Conversion of type '{ email: string; name: string; phoneNumber: string | undefined; roles: ["Admin" | "Donor" | "Mentor" | "Parent" | "Player"]; }' to type 'UpdateUserDTO' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Types of property 'roles' are incompatible.
    Type '["Admin" | "Donor" | "Mentor" | "Parent" | "Player"]' is not comparable to type '[ViewingPermissionDTO]'.
      Type 'string' is not comparable to type 'ViewingPermissionDTO'.
        Type 'string' is not comparable to type 'ViewingPermissionDTO'.`